src/imap/mod.rs

```diff
@@ -1104,7 +1104,15 @@ impl Imap {
                         });
                 info!(context, "sentbox folder is {:?}", sentbox_folder);
 
-                let delimiter = self.config.read().await.imap_delimiter;
+                let mut delimiter = ".";
+                if !folders.is_empty() {
```

Comment:
It is better to use `folders.first()` to get an `Option` instead of checking `is_empty` and indexing. `folders[0]` can panic, and it prevents use from enabling [`indexing_slicing`](https://rust-lang.github.io/rust-clippy/master/index.html#indexing_slicing) clippy lint.